diff --git testautoRIFT.py testautoRIFT.py
--- testautoRIFT.py
+++ testautoRIFT.py
@@ -541,9 +541,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
         preprocessing_methods = ['hps', 'hps']
         for ii, name in enumerate((m_name, s_name)):
             if len(re.findall("L[EO]07_", name)) > 0:
-                acquisition = datetime.strptime(name.split('_')[3], '%Y%m%d')
-                if acquisition >= datetime(2003, 5, 31):
-                    preprocessing_methods[ii] = 'wallis_fill'
+                preprocessing_methods[ii] = 'wallis_fill'
             elif len(re.findall("LT0[45]_", name)) > 0:
                 preprocessing_methods[ii] = 'fft'
 
diff --git testautoRIFT_ISCE.py testautoRIFT_ISCE.py
--- testautoRIFT_ISCE.py
+++ testautoRIFT_ISCE.py
@@ -540,9 +540,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
         preprocessing_methods = ['hps', 'hps']
         for ii, name in enumerate((m_name, s_name)):
             if len(re.findall("L[EO]07_", name)) > 0:
-                acquisition = datetime.strptime(name.split('_')[3], '%Y%m%d')
-                if acquisition >= datetime(2003, 5, 31):
-                    preprocessing_methods[ii] = 'wallis_fill'
+                preprocessing_methods[ii] = 'wallis_fill'
             elif len(re.findall("LT0[45]_", name)) > 0:
                 preprocessing_methods[ii] = 'fft'
 
