diff --git crop.py crop.py
--- a/hyp3_autorift/crop.py
+++ b/hyp3_autorift/crop.py
@@ -62,7 +62,7 @@ def crop_netcdf_product(netcdf_file: Path) -> Path:
     """
     with xr.open_dataset(netcdf_file) as ds:
         # this will drop X/Y coordinates, so drop non-None values just to get X/Y extends
-        xy_ds = ds.where(ds.v.notnull(), drop=True)
+        xy_ds = ds.where(ds.v.notnull()).dropna(dim='x', how='all').dropna(dim='y', how='all')

         x_values = xy_ds.x.values
         grid_x_min, grid_x_max = x_values.min(), x_values.max()
@@ -75,7 +75,7 @@ def crop_netcdf_product(netcdf_file: Path) -> Path:
         mask_lat = (ds.y >= grid_y_min) & (ds.y <= grid_y_max)
         mask = (mask_lon & mask_lat)

-        cropped_ds = ds.where(mask, drop=True)
+        cropped_ds = ds.where(mask).dropna(dim='x', how='all').dropna(dim='y', how='all')
         cropped_ds = cropped_ds.load()

         # Reset data for mapping and img_pair_info data variables as ds.where() extends data of all data variables