--- testGeogridOptical.py
+++ testGeogridOptical.py
@@ -100,7 +100,7 @@ def coregisterLoadMetadata(indir_m, indir_s, **kwargs):
     info.XSize = trans[1]
     info.YSize = trans[5]

-    if re.findall("L[CO]08_",DS.GetDescription()).__len__() > 0:
+    if re.findall("L[CO]0[89]_",DS.GetDescription()).__len__() > 0:
         nameString = os.path.basename(DS.GetDescription())
         info.time = nameString.split('_')[3]
     elif 'sentinel-s2-l1c' in indir_m:
@@ -120,7 +120,7 @@ def coregisterLoadMetadata(indir_m, indir_s, **kwargs):

     info1 = Dummy()

-    if re.findall("L[CO]08_",DS1.GetDescription()).__len__() > 0:
+    if re.findall("L[CO]0[89]_",DS1.GetDescription()).__len__() > 0:
         nameString1 = os.path.basename(DS1.GetDescription())
         info1.time = nameString1.split('_')[3]
     elif 'sentinel-s2-l1c' in indir_s:
--- testGeogrid_ISCE.py
+++ testGeogrid_ISCE.py
@@ -170,7 +170,7 @@ def coregisterLoadMetadataOptical(indir_m, indir_s, **kwargs):
     info.XSize = trans[1]
     info.YSize = trans[5]

-    if re.findall("L[CO]08_",DS.GetDescription()).__len__() > 0:
+    if re.findall("L[CO]0[89]_",DS.GetDescription()).__len__() > 0:
         nameString = os.path.basename(DS.GetDescription())
         info.time = nameString.split('_')[3]
     elif 'sentinel-s2-l1c' in indir_m:
@@ -190,7 +190,7 @@ def coregisterLoadMetadataOptical(indir_m, indir_s, **kwargs):

     info1 = Dummy()

-    if re.findall("L[CO]08_",DS1.GetDescription()).__len__() > 0:
+    if re.findall("L[CO]0[89]_",DS1.GetDescription()).__len__() > 0:
         nameString1 = os.path.basename(DS1.GetDescription())
         info1.time = nameString1.split('_')[3]
     elif 'sentinel-s2-l1c' in indir_s:
