--- testautoRIFT.py
+++ testautoRIFT.py
@@ -827,7 +827,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         'mission_img1': master_split[0][0],
                         'sensor_img1': 'C',
                         'satellite_img1': master_split[0][1:3],
-                        'acquisition_img1': master_dt,
+                        'acquisition_date_img1': master_dt,
                         'time_standard_img1': 'UTC',
                         'absolute_orbit_number_img1': master_split[7],
                         'mission_data_take_ID_img1': master_split[8],
@@ -836,7 +836,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         'mission_img2': slave_split[0][0],
                         'sensor_img2': 'C',
                         'satellite_img2': slave_split[0][1:3],
-                        'acquisition_img2': slave_dt,
+                        'acquisition_date_img2': slave_dt,
                         'time_standard_img2': 'UTC',
                         'absolute_orbit_number_img2': slave_split[7],
                         'mission_data_take_ID_img2': slave_split[8],
@@ -1021,11 +1021,13 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
 
                     IMG_INFO_DICT = {
                         'mission_img1': master_split[0][-3],
+                        'sensor_img1': 'MSI',
                         'satellite_img1': master_split[0][-2:],
                         'correction_level_img1': master_split[4][:3],
                         'acquisition_date_img1': master_dt,
                         'time_standard_img1': 'UTC',
                         'mission_img2': slave_split[0][-3],
+                        'sensor_img2': 'MSI',
                         'satellite_img2': slave_split[0][-2:],
                         'correction_level_img2': slave_split[4][:3],
                         'acquisition_date_img2': slave_dt,
--- testautoRIFT_ISCE.py
+++ testautoRIFT_ISCE.py
@@ -832,7 +832,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         'mission_img1': master_split[0][0],
                         'sensor_img1': 'C',
                         'satellite_img1': master_split[0][1:3],
-                        'acquisition_img1': master_dt,
+                        'acquisition_date_img1': master_dt,
                         'time_standard_img1': 'UTC',
                         'absolute_orbit_number_img1': master_split[7],
                         'mission_data_take_ID_img1': master_split[8],
@@ -841,7 +841,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         'mission_img2': slave_split[0][0],
                         'sensor_img2': 'C',
                         'satellite_img2': slave_split[0][1:3],
-                        'acquisition_img2': slave_dt,
+                        'acquisition_date_img2': slave_dt,
                         'time_standard_img2': 'UTC',
                         'absolute_orbit_number_img2': slave_split[7],
                         'mission_data_take_ID_img2': slave_split[8],
@@ -1025,11 +1025,13 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
 
                     IMG_INFO_DICT = {
                         'mission_img1': master_split[0][-3],
+                        'sensor_img1': 'MSI',
                         'satellite_img1': master_split[0][-2:],
                         'correction_level_img1': master_split[4][:3],
                         'acquisition_date_img1': master_dt,
                         'time_standard_img1': 'UTC',
                         'mission_img2': slave_split[0][-3],
+                        'sensor_img2': 'MSI',
                         'satellite_img2': slave_split[0][-2:],
                         'correction_level_img2': slave_split[4][:3],
                         'acquisition_date_img2': slave_dt,
