diff --git a/testautoRIFT.py b/testautoRIFT.py
index 22dc26b..024133b 100755
--- a/testautoRIFT.py
+++ b/testautoRIFT.py
@@ -984,14 +984,12 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         YPixelSize = geogrid_run_info['YPixelSize']
                         epsg = geogrid_run_info['epsg']
 
-                    master_path = indir_m
-                    slave_path = indir_s
+                    master_id = kwargs['reference_metadata']['id']
+                    slave_id = kwargs['reference_metadata']['id']
 
-                    master_split = kwargs['reference_metadata']['id'].split('_')
-                    slave_split = kwargs['secondary_metadata']['id'].split('_')
+                    master_split = master_id.split('_')
+                    slave_split = slave_id.split('_')
 
-                    master_filename = master_split[0][-3:]+'_'+master_split[2]+'_'+master_split[4][:3]+'_'+os.path.basename(master_path)
-                    slave_filename = slave_split[0][-3:]+'_'+slave_split[2]+'_'+slave_split[4][:3]+'_'+os.path.basename(slave_path)
 
                     master_time = ['00','00','00']
                     slave_time = ['00','00','00']
@@ -1010,7 +1008,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         raise Exception('Input search range is all zero everywhere, thus no search conducted')
                     PPP = roi_valid_percentage * 100
                     if ncname is None:
-                        out_nc_filename = f"./{master_filename[0:-8]}_X_{slave_filename[0:-8]}" \
+                        out_nc_filename = f"./{master_id}_X_{slave_id}" \
                                           f"_G{gridspacingx:04.0f}V02_P{np.floor(PPP):03.0f}.nc"
                     else:
                         out_nc_filename = f"{ncname}_G{gridspacingx:04.0f}V02_P{np.floor(PPP):03.0f}.nc"
diff --git a/testautoRIFT_ISCE.py b/testautoRIFT_ISCE.py
index 275e391..d6a6d22 100755
--- a/testautoRIFT_ISCE.py
+++ b/testautoRIFT_ISCE.py
@@ -986,14 +986,12 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         YPixelSize = geogrid_run_info['YPixelSize']
                         epsg = geogrid_run_info['epsg']
 
-                    master_path = indir_m
-                    slave_path = indir_s
+                    master_id = kwargs['reference_metadata']['id']
+                    slave_id = kwargs['reference_metadata']['id']
 
-                    master_split = kwargs['reference_metadata']['id'].split('_')
-                    slave_split = kwargs['secondary_metadata']['id'].split('_')
+                    master_split = master_id.split('_')
+                    slave_split = slave_id.split('_')
 
-                    master_filename = master_split[0][-3:]+'_'+master_split[2]+'_'+master_split[4][:3]+'_'+os.path.basename(master_path)
-                    slave_filename = slave_split[0][-3:]+'_'+slave_split[2]+'_'+slave_split[4][:3]+'_'+os.path.basename(slave_path)
 
                     master_time = ['00','00','00']
                     slave_time = ['00','00','00']
@@ -1012,7 +1010,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                         raise Exception('Input search range is all zero everywhere, thus no search conducted')
                     PPP = roi_valid_percentage * 100
                     if ncname is None:
-                        out_nc_filename = f"./{master_filename[0:-8]}_X_{slave_filename[0:-8]}" \
+                        out_nc_filename = f"./{master_id}_X_{slave_id}" \
                                           f"_G{gridspacingx:04.0f}V02_P{np.floor(PPP):03.0f}.nc"
                     else:
                         out_nc_filename = f"{ncname}_G{gridspacingx:04.0f}V02_P{np.floor(PPP):03.0f}.nc"
