diff --git testGeogrid_ISCE.py testGeogrid_ISCE.py
--- testGeogrid_ISCE.py
+++ testGeogrid_ISCE.py
@@ -27,6 +27,8 @@
 #
 # Authors: Piyush Agram, Yang Lei
 #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+from typing import Tuple
+from pathlib import Path
 
 
 def cmdLineParse():
@@ -150,6 +152,18 @@ def loadMetadata(indir,buffer=0):
 
     return info
 
+
+def get_polarizations(s1_safe: str) -> Tuple[str]:
+    mapping = {
+        'SH': ('hh',),
+        'SV': ('vv',),
+        'DH': ('hh', 'hv'),
+        'DV': ('vv', 'vh'),
+    }
+    key = Path(s1_safe).name[14:16]
+    return mapping[key]
+
+
 def loadParsedata(indir, orbit_dir, aux_dir, buffer=0):
     '''
     Input file.
@@ -170,7 +184,7 @@ def loadParsedata(indir, orbit_dir, aux_dir, buffer=0):
         rdr.orbitDir=orbit_dir
         rdr.auxDir=aux_dir
         rdr.swathNumber=swath
-        rdr.polarization='hh'
+        rdr.polarization=get_polarizations(indir)[0]
         rdr.parse()
         frames.append(rdr.product)
     
