diff --git testautoRIFT.py testautoRIFT.py
--- testautoRIFT.py
+++ testautoRIFT.py
@@ -546,7 +546,7 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
                     preprocessing_methods[ii] = 'wallis_fill'
             elif len(re.findall("LT0[45]_", name)) > 0:
                 preprocessing_methods[ii] = 'fft'
-        
+
         zero_mask = None
         indir_m_zero = f'{indir_m.split(".")[0]}_zeroMask.{indir_m.split(".")[1]}'
         indir_s_zero = f'{indir_s.split(".")[0]}_zeroMask.{indir_s.split(".")[1]}'
@@ -555,8 +555,8 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
             m_zero = m_zero.astype(np.uint8)
             s_zero = s_zero.astype(np.uint8)
 
-            # FIXME: AND? Wallis uses "or" here, while wallis_fill uses "and" here.
-            zero_mask = m_zero & s_zero
+            zero_mask = m_zero | s_zero
+            zero_mask = zero_mask.astype(np.uint8)
 
         print(f'Using preprocessing methods {preprocessing_methods}')
